src/linux/Packaging.Linux/install-from-source.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/sh
+
+SUDO_CMD=sudo
+
+# Parse script argument that allows users to customize sudo command
+for i in "$@"
+do
+    case "$i" in
+        --sudo-cmd=*)
+        SUDO_CMD="${i#*=}"
+        shift # past argument=value
+        ;;
+    esac
+done
+
+install_shared_packages() {
+    pkg_manager=$1
+    install_verb=$2
+
+    local shared_packages="git curl apt-transport-https"
+    for package in $shared_packages
+    do
+        if [ $pkg_manager = apk ]; then
+            $SUDO_CMD $pkg_manager $install_verb $package
+        else
+            $SUDO_CMD $pkg_manager $install_verb $package -y
+        fi
+    done
+}
+
+download_dotnet_script() {
+    curl -o dotnet-install.sh -L https://dot.net/v1/dotnet-install.sh
+    chmod +x ./dotnet-install.sh
+}
+
+distribution=$(sed -n '/^ID=/p' /etc/os-release | cut -c 4- | tr -d '"')
```

Comment:
A slightly more robust alternative would be:

```sh
eval "$(sed -n 's/^ID=/distribution=/p' /etc/os-release)"
```

This would not assume that the value is enclosed in double-quotes (the `cut -c 4-` assumes that the character after `ID=` is uninteresting 😄).